engine.py

```diff
@@ -260,6 +260,8 @@ def undo_move(self, look_ahead_mode = False):
             last_move = self.move_log.pop()
             self.board[last_move.start_row][last_move.start_col] = last_move.piece_moved
             self.board[last_move.end_row][last_move.end_col] = last_move.piece_captured
+            self.board[last_move.end_row + 1][last_move.end_col] = last_move.piece_captured # Undo for light en-passant
+            self.board[last_move.end_row - 1][last_move.end_col] = last_move.piece_captured # Undo for dark en-passant
```

Comment:
You neded check if the last move is an en-passant move. Because if you didn't your code will run this whole for every move

Reply:
Do something like
if last_move.move_type == "en-passent":
     # you write the undo en-passant move logic here